test_runner/performance/test_physical_replication.py

```diff
@@ -0,0 +1,232 @@
+            finally:
+                replica_workload.terminate()
+        finally:
+            master_workload.terminate()
+    except Exception:
+        should_delete = False
+    finally:
+        if should_delete:
+            neon_delete_project(neon_api_key, neon_api_base_url, project_id)
+
+
+@pytest.mark.remote_cluster
+@pytest.mark.timeout(0)
+def test_replication_start_stop(
+    pg_bin: PgBin,
+    test_output_dir: Path,
+    neon_api_key: str,
+    neon_api_base_url: str,
+    pg_version: PgVersion,
+):
+    num_replicas = 2
+    configuration_test_time_sec = 5
+    should_delete = True
+
+    project = neon_create_project(neon_api_key, neon_api_base_url, pg_version)
+    project_id = project["project"]["id"]
+    neon_wait_for_operation_to_finish(neon_api_key, neon_api_base_url, project_id)
+    try:
+        branch_id = project["branch"]["id"]
+        master_connstr = project["connection_uris"][0]["connection_uri"]
+
+        replicas = []
+        for _ in range(num_replicas):
+            replicas.append(
+                neon_create_endpoint(
+                    neon_api_key,
+                    neon_api_base_url,
+                    project_id,
+                    branch_id,
+                    endpoint_type="read_only",
+                )
+            )
+            neon_wait_for_operation_to_finish(neon_api_key, neon_api_base_url, project_id)
+
+        replica_connstr = [
+            neon_get_connection_uri(
+                neon_api_key,
+                neon_api_base_url,
+                project_id,
+                endpoint_id=replicas[i]["endpoint"]["id"],
+            )["uri"]
+            for i in range(num_replicas)
+        ]
+
+        pg_bin.run_capture(["pgbench", "-i", "-s10", master_connstr])
+        conn_master = psycopg2.connect(master_connstr)
+        cur_master = conn_master.cursor()
+
+        conn_replica: List[Optional[psycopg2.connection]] = [
+            psycopg2.connect(replica_connstr[i]) for i in range(num_replicas)
+        ]
+        cur_replica: List[Optional[psycopg2.cursor]] = []
+        for i in range(num_replicas):
+            conn = conn_replica[i]
+            assert conn is not None
+            cur_replica.append(conn.cursor())
+
+        # Sync replicas
+        for i in range(num_replicas):
+            measure_replication_lag(cur_master, cur_replica[i])
+
+        master_pgbench = pg_bin.run_nonblocking(
+            [
+                "pgbench",
+                "-c10",
+                "-T1000",
+                "-Mprepared",
+                "--log",
+                f"--log-prefix={test_output_dir}/pgbench_master",
+                f"--aggregate-interval={configuration_test_time_sec}",
+                master_connstr,
+            ]
+        )
+        replica_pgbench: List[Optional[subprocess.Popen[Any]]] = [None for i in range(num_replicas)]
+
+        for iconfig in range((1 << num_replicas) - 1, -1, -1):
```

Comment:
I'm not sure that I understood why we need to run it `num_replicas^2` times; probably we need a comment here, as well?

Reply:
Sure. It's just that we want to test each configuration, with each read replica being either on or off